modules/test/playwright/tests/export-import-service/main/remoteStaging.spec.ts

```diff
@@ -45,6 +46,332 @@ export const test = mergeTests(
+					titleMap: {en_US: webContentTitle},
+				});
+
+				const structureName2 = getRandomString();
+				const dataDefinition2 = getDataStructureDefinition({
+					defaultLanguageId: 'en_US',
+					fields: [
+						{name: 'Content1', repeatable: false},
+						{name: 'Content2', repeatable: false},
+					],
+					name: structureName2,
+				});
+
+				const structure2 = await apiHelpers.dataEngine.createStructure(
+					site.id,
+					dataDefinition2
+				);
+
+				const templateName2 = 'template2';
+				const templateScript2 =
+					'<h1>${Content1.getData()}</h1>\n' +
+					'<p>${Content2.getData()}</p>';
+
+				await journalEditTemplatePage.goto(site.friendlyUrlPath);
+				await journalEditTemplatePage.selectStructure(structureName2);
+				await journalEditTemplatePage.editTemplate(
+					templateName2,
+					templateScript2
+				);
+				await journalEditTemplatePage.saveTemplate();
+				await journalEditTemplatePage.selectTemplateToEdit(
+					templateName2
+				);
+
+				const templateKey2 =
+					await journalEditTemplatePage.getDDMTemplateKey();
+
+				await webContentDisplayPage.gotoWebContentAdmin(site.name);
+
+				for (const num of pageNumbers) {
+					await apiHelpers.jsonWebServicesJournal.addWebContent({
+						contentFields: [
+							{name: `Content1`, value: `Title-${num}`},
+							{
+								name: `Content2`,
+								value: `Text Content-${num}`,
+							},
+						],
+						ddmStructureId: structure2.id,
+						ddmTemplateKey: templateKey2,
+						groupId: site.id,
+						titleMap: {en_US: `Title-${num}`},
+					});
+
+					await reloadUntilVisible({
+						myLocator: page.getByRole('link', {
+							name: `Title-${num}`,
+						}),
+						page,
+					});
+				}
+			});
+
+			let i = 0;
+			await test.step('Add web content to pages', async () => {
+				for (const layout of layouts) {
+					await pageEditorPage.goto(layout, site.friendlyUrlPath);
+					let attempt = 0;
+					do {
+						attempt++;
+						await page.reload({waitUntil: 'domcontentloaded'});
+
+						if (
+							await page
+								.getByText(webContentTitle, {exact: true})
+								.isVisible()
+						) {
+							break;
+						}
+						await webContentDisplayPage.addWebContentWithDisplay({
+							pageType: 'content',
+							waitAfterAddingWebcontent: true,
+							webContentName: webContentTitle,
+						});
+						await page.waitForTimeout(500);
+					} while (attempt <= 100);
+					attempt = 0;
+					do {
+						attempt++;
+						await page.reload({waitUntil: 'domcontentloaded'});
+						if (
+							await page
+								.getByText(`Title-${pageNumbers[i]}`, {
+									exact: true,
+								})
+								.first()
+								.isVisible()
+						) {
+							break;
+						}
+						await webContentDisplayPage.addWebContentWithDisplay({
+							pageType: 'content',
+							waitAfterAddingWebcontent: true,
+							webContentName: `Title-${pageNumbers[i]}`,
+						});
+
+						await page.waitForTimeout(500);
+					} while (attempt <= 100);
+
+					i++;
+				}
```

Comment:
This code still looks odd. The `attempt` handling and the `await page.waitForTimeout(500);` are something we should avoid. 

Have you tried the `waitFor()` or any other mechanism?